src/dns_internal.cc

```diff
@@ -930,7 +930,20 @@ idnsSendQueryVC(idns_query * q, size_t nsn)
         return;
     }
 
-    vc->queue->reset();
+    if (vc->queue->isNull())
+        vc->queue->init();
+
+    const auto serialiedQuerySize = 2 + q->sz + 1; // payload_length + payload + terminate()
```

Comment:
We can avoid this magic constant `2` if we declare/initialize `head` _before_ checking this overflow condition. Please let me know if you prefer to split `head` initialization and append() calls that way, and I would be happy to do it despite the diff noise.